backend/src/core/services/phonebook.service.ts

```diff
@@ -0,0 +1,46 @@
+import { loggerWithLabel } from '@core/logger'
+import { ChannelType } from '@core/constants'
+import PhonebookClient from '@shared/clients/phonebook-client.class'
+import config from '@core/config'
+import { User } from '@core/models'
+
+const logger = loggerWithLabel(module)
+
+const phonebookClient: PhonebookClient = new PhonebookClient(
+  config.get('phonebook.endpointUrl'),
+  config.get('phonebook.apiKey'),
+  config.get('phonebook.version')
+)
+
+const getPhonebookLists = async ({
+  userId,
+  channel,
+}: {
+  userId: number
+  channel: ChannelType
+}): Promise<{ id: number; name: string }[]> => {
+  const user = await User.findOne({
+    where: { id: userId },
+    attributes: ['email'],
+  })
+  if (!user) {
+    logger.error('invalid user making request!')
+    return []
+  }
+  const res = await phonebookClient.getManagedLists(user.email, channel)
```

Comment:
This should handle the phonebook api errors. 4xx or 5xx should not be thrown to user rather a message like "lists cannot be retrieved" or its equivalent.
Mainly because this request is constructed and api is not invoked by postman not by the user

Reply:
to handle in controller / middleware level

Reply:
for this endpoint, i will be returning a 200 (empty list) as there is technically no list to select from, ignoring the internal error but logging the error on DD instead.